### PR TITLE
Simplify instance actions

### PR DESCRIPTION
**Pull request by edlerd**
- use portal to set actions on instance detail page
 - close instance edit stays on current page
 - extract create snapshot btn

### CODE REVIEW

File: static/js/buttons/instances/CreateSnapshotBtn.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  instance?: LxdInstance;
+  notify: NotificationHelper;
+}
+
+const CreateSnapshotBtn: FC<Props> = ({ instance, notify }) => {
+  const [isModal, setModal] = useState(false);
+
+  return (
+    <>
+      {instance && isModal && (
```
Comment (lorumic) [inline static/js/buttons/instances/CreateSnapshotBtn.tsx:17]: I think checking that `instance` is not `undefined` should not be done here, as we shouldn't even have this rendered if `instance` is not defined. We also do the same in other places I think, check in the parent and then pass a clean, "defined" prop to the children.

File: static/js/InstanceOverview.tsx [hunk 2]
```diff
@@ -93,10 +48,31 @@ const InstanceOverview: FC<Props> = ({ instanceName, setControls }) => {
...
+      {controlTarget &&
+        createPortal(
+          <>
+            <StartInstanceBtn label="Start" {...btnProps} />
+            <StopInstanceBtn label="Stop" {...btnProps} />
+            <EditInstanceBtn label="Edit" {...btnProps} />
+            <DeleteInstanceBtn
+              onFinish={() => navigate("/instances")}
+              {...btnProps}
+            />
```
Comment (lorumic) [inline static/js/InstanceOverview.tsx:71]: This is missing the `label="Delete"` prop.

File: static/js/modals/CreateSnapshotForm.tsx [hunk 3]
```diff
@@ -79,28 +79,25 @@ const CreateSnapshotForm: FC<Props> = ({ instance, close, notify }) => {
...
+      buttonRow={
+        <>
+          <Button className="u-no-margin--bottom" type="button" onClick={close}>
+            Cancel
+          </Button>
+          <SubmitButton
+            isSubmitting={formik.isSubmitting}
+            isDisabled={!formik.isValid}
+            buttonLabel="Create snapshot"
+            onClick={() => void formik.submitForm}
```
Comment (lorumic) [inline static/js/modals/CreateSnapshotForm.tsx:94]: Why the `Modal` outside the `Form`? We could have avoided the redirection of the `click` event.

Comment (edlerd) [reply static/js/modals/CreateSnapshotForm.tsx:94]: It creates a visual bug, the CreateSnapshot Button goes into its own row when the modal is wrapped in a form.

**Pull request merged**